### PR TITLE
Role-based auth

### LINKED ISSUE #82: Add permissions to roles on Auth0

### PR DESCRIPTION
**Pull request by hangler**
Fixes #82. Includes an example of gating access to a route (`/insertEntry`) using `redirectIfUserLacksPermission(...)`.

### CODE REVIEW

File: app/routes/bank/index.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { PageHeader } from "~/components/elements/PageHeader"
+import { redirectIfUserNotLoggedIn } from "~/services/auth/session.server"
+
+export const loader = async (request: Request) => {
+  await redirectIfUserNotLoggedIn(request)
```
Comment (Shawk031) [inline app/routes/bank/index.tsx:5]: for some reason, when I navigate directly to http://localhost:3000/bank, this line throws an error. Specifically, the error message is

```
TypeError: Cannot read properties of undefined (reading 'get')
    at getCookieSession (C:\Users\user\Documents\Work\dchp3\app\services\auth\session.server.ts:28:71)...
```

File: app/routes/insertEntry.tsx [hunk 2]
```diff
@@ -12,6 +14,12 @@ export async function action({ request }: ActionArgs) {
   return redirect(`/entries/${data.headword}`)
 }
 
+export async function loader({ request }: LoaderArgs) {
+  await redirectIfUserLacksPermission(request, "det:createDraft")
```
Comment (Shawk031) [inline app/routes/insertEntry.tsx:18]: So I think we haven't set up permissions for me here yet, so I can't test this part. But the redirect works at least haha

**Review by Shawk031 — APPROVED** [review]
Looks good! The only thing is the redirect from /bank throwing that error. I'd be down for a separate PR fix though

File: app/routes/insertEntry.tsx (hunk 2 above)
Comment (hangler) [reply app/routes/insertEntry.tsx:18]: You should now have admin privileges. You might need to log out (go to the Admin page to do this if necessary) and log back in. Let me know how it goes!

File: app/routes/bank/index.tsx (hunk 1 above)
Comment (hangler) [reply app/routes/bank/index.tsx:5]: Thanks for the catch. Dumb error.

File: app/routes/insertEntry.tsx (hunk 2 above)
Comment (Shawk031) [reply app/routes/insertEntry.tsx:18]: The access works by the way. Thanks!

**Pull request merged**